BotRoyaleUnity/Assets/Scripts/buildRobot.cs

```diff
@@ -7,57 +7,89 @@
 
 public class buildRobot : MonoBehaviour
 {
-    public GameObject block,center,spike;
-    public Vector3 pos,rot;
-    
-
-// Start is called before the first frame update
-void Start()
-    {   
-        string jsonString = File.ReadAllText("Assets/Scripts/test.json");
-        var N = JSON.Parse(jsonString);
-        string test = N[0]["type"];
-        int a = 0;
-        while(N[a] != null){
-            pos = new Vector3((float)N[a]["x"],0.5f,(float)N[a]["y"]);
-            string type = N[a]["type"];
-            Quaternion rot;
-            string direcction = N[a]["direction"];
-            switch(direcction){
-                case "north":
-                    rot = Quaternion.Euler(90,0,0);
-                    break;
-                case "south":
-                    rot = Quaternion.Euler(-90,0,0);
-                    break;
-                case "east":
-                    rot = Quaternion.Euler(0,0,-90);
-                    break;
-                case "west":
-                    rot = Quaternion.Euler(0,0,90);
-                    break;
-                default:
-                    rot = Quaternion.Euler(0,0,0);
-                    break;
+    public GameObject block, center, spike;
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        string jsonString = File.ReadAllText("Assets/Scripts/robot1.json");
+        GameObject robot1  = build(jsonString);
+        robot1.AddComponent<RoombaMovement>();
+        jsonString = File.ReadAllText("Assets/Scripts/robot2.json");
+        GameObject robot2  = build(jsonString);
+        robot2.AddComponent<RoombaMovement>();
+    }
+
+    void setParent(GameObject parent, GameObject child){
+        child.transform.parent = parent.transform;
+    }
+
+    GameObject build(string jsonString){
+        var json = JSON.Parse(jsonString);
+        int index = 0;
+        GameObject centerParent = null;
+        while (json[index] != null && centerParent == null)
```

Comment:
The center block should maybe be a child of the root parent like all the other parts. Collision detection with nested colliders can lead to child colliders causing events on the parent to happen. Since we'll be detecting collisions on each robot part seperatly, the center block probably shouldn't be a parent of all the other parts